File: Sorts/SelectionSort/selectionSort.swift
func selectionSort(_ arr: [Int]) -> [Int] {
    var sortedArr = arr
    let len = sortedArr.count
    for i in 0..<len {
        var minIndex = i
        for j in (i + 1)..<len {
            if sortedArr[j] < sortedArr[minIndex] {
                minIndex = j
            }
        }
        if minIndex != i {
            let temp = sortedArr[i]
            sortedArr[i] = sortedArr[minIndex]
            sortedArr[minIndex] = temp
        }
    }
    return sortedArr
}

var arr = [64, 34, 25, 12, 22, 11, 90]
let sortedArr = selectionSort(arr)
print(sortedArr)

File: Sorts/SelectionSort/selectionSort.go
package main

import "fmt"

func selectionSort(arr []int) []int {
    len := len(arr)
    for i := 0; i < len; i++ {
        minIndex := i
        for j := i + 1; j < len; j++ {
            if arr[j] < arr[minIndex] {
                minIndex = j
            }
        }
        if minIndex > i {
            temp := arr[i]
            arr[i] = arr[minIndex]
            arr[minIndex] = temp
        }
    }

    return arr
}

func main() {
    arr := []int{64, 34, 25, 12, 22, 11, 90}
    sortedArr := selectionSort(arr)
    fmt.Println(sortedArr)
}